rust/macros/module.rs

```diff
@@ -111,19 +106,14 @@ impl<'a> ModInfoBuilder<'a> {
 }
 
 fn permissions_are_readonly(perms: &str) -> bool {
-    let (radix, digits) = if let Some(n) = perms.strip_prefix("0x") {
-        (16, n)
-    } else if let Some(n) = perms.strip_prefix("0o") {
-        (8, n)
-    } else if let Some(n) = perms.strip_prefix("0b") {
-        (2, n)
-    } else {
-        (10, perms)
+    // A non-panicing try_split_at()
+    let (code, radix) = match perms.get(0..2).zip(perms.get(2..)) {
```

Comment:
I'm not sure this is really more readable.

Reply:
a later commit makes it one line here and moves it to helpers.rs. but I can do this without the .zip() which may be better

Reply:
I removed this as well via rebase.